### ISSUE TITLE
Triton cannot inference `tf.math.l2_normalize` correctly from ngc 21.06 ~ ngc 22.03 ( triton 2.20.0)

### ISSUE THREAD

**Issue by kimdwkimdw** [triton-inference-server/server#3978]
**Description**

I've converted layers from tensorflow to TensorRT.
This model's result is valid until ngc 21.05.
From 21.06 to 22.01, TensorRT models containing layers converted from `tf.math.l2_normalize` cause wrong result in Triton inference server.

Expected values are in range between 0.0~1.0, but it generates wrong values from `1e+2` to `1e+5` like exploding.

I've get correct result without Triton inference with script(https://gist.github.com/kimdwkimdw/2188b4b9ff9f9bf2f7e8dbd999769796)

reference: https://github.com/NVIDIA/TensorRT/issues/1707

**Triton Information**

NGC tags until 21.05-py: valid result
NGC tags from 21.06-py to 22.01-py3: wrong result

https://catalog.ngc.nvidia.com/orgs/nvidia/containers/tritonserver

**To Reproduce**
Steps to reproduce the behavior.

Describe the models (framework, inputs, outputs), ideally include the model configuration file (if using an ensemble include the model configuration file for that as well).

Any network from tensorflow including `tf.math.l2_normalize` operation converting from ONNX OPSET from 11 to 13. 
Next, convert onnx model to TensortRT model with tensorrt with 21.06 to 22.01(https://catalog.ngc.nvidia.com/orgs/nvidia/containers/tensorrt)

**Expected behavior**

Expected values are in range between 0.0~1.0, but it generates wrong values from `1e+2` to `1e+5` like exploding.

**Comment by dyastremsky** [comment]
Thank you for bringing this to our attention. We've filed a ticket to investigate.

**Comment by cnegron-nv** [comment]
@kimdwkimdw This does appear to be a TRT related issue, have you by chance asked a this question from the TRT team? https://github.com/NVIDIA/TensorRT/issues

**Comment by kimdwkimdw** [comment]
> @kimdwkimdw This does appear to be a TRT related issue, have you by chance asked a this question from the TRT team? https://github.com/NVIDIA/TensorRT/issues

I've already checked with TensorRT. https://github.com/NVIDIA/TensorRT/issues/1707

But it is appeared in only in Triton inference server.

**Comment by cnegron-nv** [comment]
@kimdwkimdw The bug referenced is about an ONNX model, not sure how it relates.

**Comment by kimdwkimdw** [comment]
> @kimdwkimdw The bug referenced is about an ONNX model, not sure how it relates.

When I convert `tf.math.l2_normalize` with fp32, triton inference works well.

But, I thought that this issue is related with fp16.
With fp16, trt(TensorRT) model without triton inference works well.
However, I tried this model with triton inference and TensorRT backend, it's result is  exploding.

**Comment by cnegron-nv** [comment]
@kimdwkimdw Can you run the model with trtexec and see if it runs correctly in this case?

**Comment by kimdwkimdw** [comment]
> @kimdwkimdw Can you run the model with trtexec and see if it runs correctly in this case?

I've tested and checked my `model.plan` runs correctly with Python API (https://gist.github.com/kimdwkimdw/2188b4b9ff9f9bf2f7e8dbd999769796) (https://docs.nvidia.com/deeplearning/tensorrt/developer-guide/index.html#python_topics)

This `model.plan` file doesn't run correctly in triton inference.

**Comment by yuanyao-nv** [comment]
I created a model with `l2_normalize()` and it seems to run correctly with triton.

Here are my steps:

Create TF model:
```
class L2NormalizeLayer(keras.layers.Layer):
    def __init__(self):
        super(L2NormalizeLayer, self).__init__(dtype=tf.float16)

    def call(self, inputs):
        return tf.math.l2_normalize(inputs)

x = keras.Input(shape=(128,),dtype='float16')
y = L2NormalizeLayer()(x)
model = keras.Model(x,y)

model.save('L2NormalizeModel')
```

Convert TF model to ONNX:
`$ pip install -U tf2onnx`
`$ python3 -m tf2onnx.convert --saved-model L2NormalizeModel/ --output L2NormalizeModel.onnx --opset 11`
 
Create TRT engine:
`$ trtexec --onnx=L2NormalizeModel.onnx --saveEngine=model.plan --minShapes=input_1:1x128 --maxShapes=input_1:1x128 --optShapes=input_1:1x128`

And then run tritonserver and the client.

My client script is based on [this example](https://github.com/triton-inference-server/client/blob/main/src/python/examples/simple_grpc_infer_client.py) with the following modifications:
```
    model_name = "L2Normalizer"

    # Infer
    inputs = []
    outputs = []
    inputs.append(grpcclient.InferInput('input_1', [1,128], "FP16"))

    input1_data = np.random.rand(1,128).astype(np.float16)

    # Initialize the data
    inputs[0].set_data_from_numpy(input1_data)

    outputs.append(grpcclient.InferRequestedOutput('l2_normalize_layer'))

    # Test with outputs
    results = triton_client.infer(
        model_name=model_name,
        inputs=inputs,
        outputs=outputs,
        client_timeout=FLAGS.client_timeout,
        headers={'test': '1'},
        compression_algorithm=FLAGS.grpc_compression_algorithm)

    statistics = triton_client.get_inference_statistics(model_name=model_name)
    print(statistics)
    if len(statistics.model_stats) != 1:
        print("FAILED: Inference Statistics")
        sys.exit(1)

    # Get the output arrays from the results
    output1_data = results.as_numpy('l2_normalize_layer')
```

@kimdwkimdw Can you try these steps again or provide your model file and/or scripts?

**Comment by kimdwkimdw** [comment]
@yuanyao-nv 

https://gist.github.com/kimdwkimdw/28fb456d15f1f21111345ea57771179a

Here are example.

1. Generate saved_model first. [generate_saved_model.py](https://gist.github.com/kimdwkimdw/28fb456d15f1f21111345ea57771179a#file-generate_saved_model-py)
2. Convert it to "ort(onnx) model"
3. Convert it to TensortRT plan with `trtexec` with fp16
```
trtexec --onnx={onnx_model}
        --saveEngine={plan_path}
        --workspace=4096
        --minShapes={MIN_SHAPES}
        --optShapes={OPT_SHAPES}
        --maxShapes={MAX_SHAPES}
        --fp16
```
4. Run `tritonserver` in NGC 22.01 ( nvcr.io/nvidia/tritonserver:22.01-py3 )
5. run_test_triton.py (https://gist.github.com/kimdwkimdw/28fb456d15f1f21111345ea57771179a#file-run_test_triton-py)

It will explode to `inf`.

**Comment by kimdwkimdw** [comment]
@yuanyao-nv @dyastremsky 
22.03 have a problem too.

**Comment by dyastremsky** [comment]
Thank you for the update. @yuanyao-nv is investigating.

(@cnegron-nv, I'm leaving this ticket open for now. Feel free to update me or the ticket if it should be closed, since the issue is happening within TensorRT and there's an open TensorRT GitHub issue for it.)

**Comment by yuanyao-nv** [comment]
Thanks for waiting. This should be fixed now in TRT 8.5.

**Comment by dyastremsky** [comment]
Wonderful! Thanks Yuan.

I'm going to close this ticket, since it's fixed on the TRT side.
